lib/dato/uploads.rb

```diff
@@ -49,6 +51,22 @@ def retrieve_job_result(job_id)
       headers.get("https://site-api.datocms.com/job-results/#{job_id}")
     end
 
+    def poll_job_result(job_id)
+      attempts = 0
+
+      while attempts < MAX_JOB_POLLING_RETRIES
+        begin
+          response = retrieve_job_result(job_id).parse
+          upload_id = response["data"]["attributes"]["payload"]["data"]["id"]
+          return upload_id
+        rescue
+          attempts += 1
+          puts "Upload job retrieval failed, trying again (ATTEMPT #{attempts} of #{MAX_JOB_POLLING_RETRIES})"
+          sleep(attempts) # exponential by default
```

Comment:
exponential would mean, 1,2,4,8. But you are going linear 1,2,3,4 😛

Reply:
and if you are using an upper limit `MAX_JOB_POLLING_RETRIES`, there is no need for a while

Rather use:
`MAX_JOB_POLLING_RETRIES.times { |i| puts "hello #{i}" }`

Reply:
Silly me

Reply:
Completely forgot about iterators in ruby!
Seems like I don't use them that often.

Reply:
Is it best (for the sleep statement) to use constant time, linear or exponential?

Reply:
Yeah, a test would be great, but would it work for third parties? Do any comon Dato models exist? If Dato-Rails were a internal project, I wouldn't hesitate, but if it can be used by the public, then its a different scenario.

Reply:
> Is it best (for the sleep statement) to use constant time, linear or exponential?

i think linear is ok